state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2046,12 +2047,29 @@ pub mod pallet {
 		) {
 			let is_ccm_swap = maybe_ccm_metadata.is_some();
 
+			let aux_lookup_key = maybe_ccm_metadata.as_ref().and_then(|ccm| {
+				T::CcmValidityChecker::decode_unchecked(
```

Comment:
We only test decode_unchecked returns Ok() - we should probably check the data is correct too in some of the tests?

Reply:
Yeah it is safe to decode without checks here because the check would have been done for all cases when the swap was initiated. We do this check when deposit channels are open and when Vault swaps are processed. So by the time we are here - we can be sure that the metadata is valid.

Reply:
Sorry, wasn't super clear, I mean to add actual unit tests for it, nothing to change in this code here :)

Reply:
Sorry I misread your comment. I've now added a unit test for this